presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -360,38 +367,88 @@ protected Object visitSearchedCaseExpression(SearchedCaseExpression node, Object
             return result;
         }
 
+        private Object processWithExceptionHandling(Expression expression, Object context)
+        {
+            if (expression == null) {
+                return null;
+            }
+
+            try {
+                return process(expression, context);
+            }
+            catch (RuntimeException e) {
+                // HACK
+                // Certain operations like 0 / 0 or likeExpression may throw exceptions.
+                // Wrap them a FunctionCall and add it to the expression evaluation
```

Comment:
`Wrap them a FunctionCall that will throw the exception if the expression is actually executed`